src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -943,6 +949,43 @@ public static String slurp(final InputStream is, final Charset charSet) {
 
         return output;
     }
+
+    /**
+     * Check if the given URI has a scheme.
+     *
+     * @param uriString the URI to check
+     * @return <code>true</code> if the given URI has a scheme, <code>false</code> if
+     * not, or if the URI is malformed.
+     */
+    public static boolean hasScheme(String uriString) {
+        try {
+            return new URI(uriString).getScheme() != null;
+        } catch (URISyntaxException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Converts the given URI to a {@link Path} object. If the filesystem cannot be found in the usual way, then attempt
+     * to load the filesystem provider using the thread context classloader. This is needed when the filesystem
+     * provider is loaded using a URL classloader (e.g. in spark-submit).
+     *
+     * @param uriString the URI to convert
+     * @return the resulting {@code Path}
+     * @throws IOException an I/O error occurs creating the file system
+     */
+    public static Path getPath(String uriString) throws IOException {
```

Comment:
could you replace the use of `Paths.get` in [`SamInputResource.UrlInputResource`](https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/SamInputResource.java#L351) with a call to this instead?  Just in case...